### PR TITLE
Update code formatting and fix typos

### LINKED ISSUE #1524: Documentation: Native english speaker validation
## Requirement

As a main contributor to GO Feature Flag, I am also the main contributor to the documentation website https://docs.gofeatureflag.org.

Since I am not a native English speaker _(sorry I am 🇫🇷)_, the English level is not perfect and may be improved.
I am looking for contributors ready to help the documentation to reach the next level.

The source of the documentation is available at https://github.com/thomaspoignant/go-feature-flag/tree/main/website/docs.

### PR DESCRIPTION
**Pull request by D5D0Ds**
# Description
This pull request aims to improve the English quality of the GO Feature Flag documentation. Recognizing the importance of clear and correct language in technical documentation, this PR introduces enhancements to various sections of the documentation.

# Changes include
- [x] Bugfix (non-breaking change that solves an issue) - Specifically, addressing linguistic improvements in the documentation.

# Closes issue(s)
This pull request aims to address the need for native English speaker validation of the documentation.

Resolve #1524

# Checklist
- [x] I have tested this code
- [x] No new unit tests required, but the updated documentation has been meticulously proofread.
- [x] I have updated the documentation (`README.md` and `/website/docs`)
- [x] I have followed the [contributing guide](CONTRIBUTING.md)

### CODE REVIEW

File: website/docs/configure_flag/rule_format.md [hunk 1]
```diff
@@ -74,20 +74,20 @@ A rule is a configuration that allows to serve a variation based on some conditi
 
 
 :::info
-`variation`, `percentage` and `progressiveRollout` are optional but you **must have one of the 3**.
+`variation`, `percentage` and `progressiveRollout` are optional but you **must have atleast one of the three**.
```
Comment (thomaspoignant) [inline website/docs/configure_flag/rule_format.md:77]:
```suggestion
`variation`, `percentage` and `progressiveRollout` are optional but you **must have at least one of the three**.
```

File: website/docs/go_module/data_collection/google_cloud_storage.md [hunk 2]
```diff
@@ -6,16 +6,16 @@ sidebar_position: 2
 
 The **Google Cloud Storage exporter** will collect the data and create a new file in a specific folder everytime we send the data.
 
-Everytime the `FlushInterval` or `MaxEventInMemory` is reached a new file will be added to S3.
+Everytime the `FlushInterval` or `MaxEventInMemory` is reached, a new file will be added to Google Cloud Storage.
```
Comment (thomaspoignant) [inline website/docs/go_module/data_collection/google_cloud_storage.md:9]: Good catch here :+1:

**Review by thomaspoignant — APPROVED** [review]
@D5D0Ds This is awesome all the things you've spotted.
I have not much to say about it except that I am super happy about all your changes.

Let me know if you are done with this PR and I will merge it. If you still want to work on it let me know too so I'll wait before merging.

**Comment by D5D0Ds** [comment]
Thank you so much for your positive feedback! I'm glad to hear that you're happy with the changes. I've reviewed the PR and everything looks good to me. You can go ahead and merge it. If there are any further tweaks or adjustments you'd like to make, just let me know and I'm more than happy to assist. @thomaspoignant . Also, I have adjusted according to the suggested changes in my latest commit.

**Pull request merged**